logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageRepository.kt

```diff
@@ -88,15 +92,17 @@ class MessageDataSource(
         return Either.Right(Unit)
     }
 
-    override suspend fun getMessageById(conversationId: ConversationId, messageUuid: String): Either<CoreFailure, Message> {
-        //TODO handle failures
-        return Either.Right(
+    override suspend fun getMessageById(conversationId: ConversationId, messageUuid: String): Either<CoreFailure, Message> =
+        wrapStorageRequest {
             messageDAO.getMessageById(messageUuid, idMapper.toDaoModel(conversationId))
-                .filterNotNull()
-                .map(messageMapper::fromEntityToMessage)
-                .first()
-        )
-    }
+                .firstOrNull()?.run {
+                    messageMapper.fromEntityToMessage(this)
+                }
+        }.onSuccess {
+            Either.Right(it)
+        }.onFailure {
+            Either.Left(StorageFailure.DataNotFound)
```

Comment:
I think that `wrapStorageRequest` can also return other failures - should we also handle them as a `DataNotFound`?

Reply:
I changed it to just forward it the same `StorageFailure` failure returned by the `messageDao` instance